### ISSUE TITLE
Failed to select an OpenTelemetry Collector instance for pod's sidecar 

### ISSUE THREAD

**Issue by gerijani** [open-telemetry/opentelemetry-operator#2548]
### Describe the issue you're reporting

Following [this issue](https://github.com/open-telemetry/opentelemetry-operator/issues/2546) -which was closed impatiently - I'd just like to understand why the default HELM chart deployment of a collector is not useful in this context? 
https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/values.yaml

$ kubectl get pods -n opentelemetry
NAME READY STATUS RESTARTS AGE
collector-0 1/1 Running 0 2d19h
operator-6479d99d-s4nk8 2/2 Running 1 (20h ago) 20h

Collector instance was deployed as statefuleset yet, we get this error:
```
{"level":"error","ts":"2024-01-19T10:32:29Z","msg":"failed to select an OpenTelemetry Collector instance for this pod's sidecar",
"namespace":"opentelemetry",
"name":"platform-samples-api-2",
"error":"no OpenTelemetry Collector instances available",
"stacktrace":
"github.com/open-telemetry/opentelemetry-operator/pkg/sidecar.(*sidecarPodMutator).Mutate
/home/runner/work/opentelemetry-operator/opentelemetry-operator/pkg/sidecar/podmutator.go:84\ngithub.com/open-telemetry/opentelemetry-operator/internal/webhook/podmutation.(*podMutationWebhook).Handle
/home/runner/work/opentelemetry-operator/opentelemetry-operator/internal/webhook/podmutation/webhookhandler.go:92\nsigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).Handle
/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/webhook/admission/webhook.go:169\nsigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).ServeHTTP
/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/webhook/admission/http.go:98\nsigs.k8s.io/controller-runtime/pkg/webhook/internal/metrics.InstrumentedHook.InstrumentHandlerInFlight.func1
/home/runner/go/pkg/mod/github.com/prometheus/client_golang@v1.18.0/prometheus/promhttp/instrument_server.go:60\nnet/http.HandlerFunc.ServeHTTP
/opt/hostedtoolcache/go/1.21.5/x64/src/net/http/server.go:2136\ngithub.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerCounter.func1
/home/runner/go/pkg/mod/github.com/prometheus/client_golang@v1.18.0/prometheus/promhttp/instrument_server.go:147\nnet/http.HandlerFunc.ServeHTTP
/opt/hostedtoolcache/go/1.21.5/x64/src/net/http/server.go:2136\ngithub.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerDuration.func2
/home/runner/go/pkg/mod/github.com/prometheus/client_golang@v1.18.0/prometheus/promhttp/instrument_server.go:109\nnet/http.HandlerFunc.ServeHTTP
/opt/hostedtoolcache/go/1.21.5/x64/src/net/http/server.go:2136\nnet/http.(*ServeMux).ServeHTTP
/opt/hostedtoolcache/go/1.21.5/x64/src/net/http/server.go:2514\nnet/http.serverHandler.ServeHTTP
/opt/hostedtoolcache/go/1.21.5/x64/src/net/http/server.go:2938\nnet/http.(*conn).serve
/opt/hostedtoolcache/go/1.21.5/x64/src/net/http/server.go:2009"}
```

Basically, we'd like to automate deployment without any manual intervention

**Comment by pavolloffay** [comment]
@gerijani I think the issue should be opened against the helm chart project. The helm chart manages which objects (e.g. OTEL collector CR as sidecar) should be created.

**Comment by iblancasa** [comment]
#2558

**Comment by pavolloffay** [comment]
Duplicate of #2558
